src/Friend.ts

```diff
@@ -0,0 +1,10 @@
+import Peer from "peerjs";
+
+export interface IFriend {
+    readonly id: string;
+    readonly displayName: string;
+}
+
+export interface IFriendConnection extends IFriend {
+    readonly connection?: Peer.DataConnection;
```

Comment:
I think we can remove the optional-ness to this. Since we won't add the friend to `this.#friends` until we make the connection.

Reply:
yeah connection is optional in host connection design but I think we're moving to everyone being connected

Reply:
It's still optional because connections are async and also we're now storing the current user in the map as well (for good reason)